src/monaca.js

```diff
@@ -2014,6 +2014,29 @@
       );
   };
 
+  Monaca.prototype._npmInit = function() {
+    if (npm) {
+      return Q.resolve(npm);
+    }
+
+    var npmModules = ['npm', 'global-npm'];
+
+    for (var i in npmModules) {
+      try {
+        npm = require(npmModules[i]);
+      } catch (err) {
+        if (err.code === 'MODULE_NOT_FOUND' && i == (npmModules.length - 1)) {
+          err.message = 'npm module not found, add it in order to be able to use this functionality.';
+          return Q.reject(err);
+        }
+      }
+
+      if (npm) {
+        return Q.resolve(npm);
+      }
+    }
+  }
+
```

Comment:
There is a possibility that the function returns without any resolve/reject. Please fix that, and it can be merged to `master`.